src/c_kzg_4844.c

```diff
@@ -2773,83 +2773,116 @@ static const fr_t INV_SCALE_FACTOR = {
 };
 
 /**
- * Scale a polynomial in place.
+ * Shift a polynomial in place.
  *
- * Multiplies each coefficient by `1 / scale_factor ^ i`. Equivalent to
- * creating a polynomial that evaluates at `x * k` rather than `x`.
+ * Multiplies each coefficient by `shift_factor ^ i`. Equivalent to
+ * creating a polynomial that evaluates at `x * shift_factor` rather than `x`.
  *
- * @param[in,out]   p       The polynomial coefficients to be scaled
- * @param[in]       len     Length of the polynomial coefficients
+ * @param[in,out]   p            The polynomial coefficients to be scaled
+ * @param[in]       len          Length of the polynomial coefficients
+ * @param[in]       shift_factor Shift factor
  */
-static void scale_poly(fr_t *p, size_t len) {
+static void shift_poly(fr_t *p, size_t len, const fr_t *shift_factor) {
     fr_t factor_power = FR_ONE;
     for (size_t i = 1; i < len; i++) {
-        blst_fr_mul(&factor_power, &factor_power, &INV_SCALE_FACTOR);
+        blst_fr_mul(&factor_power, &factor_power, shift_factor);
         blst_fr_mul(&p[i], &p[i], &factor_power);
     }
 }
 
-/**
- * Unscale a polynomial in place.
+/** Do an FFT over a coset of the roots of unity. The function mutates the
+ *  input array `in`
```

Comment:
Also, I don't like that `in` is mutated. I understand it's because of the initial shift. I know it doesn't really matter, because `extended_evaluation_times_zero_coeffs` (the only thing that's impacted) isn't used again.

One simple solution would be to `shift_poly(in, n, &INV_SCALE_FACTOR)` at the bottom.